apps/cowswap-frontend/src/modules/trade/hooks/useNotifyWidgetTrade.ts

```diff
@@ -14,10 +14,17 @@ import { TradeDerivedState } from '../types/TradeDerivedState'
 
 export function useNotifyWidgetTrade() {
   const state = useDerivedTradeState()
+  const isFirstLoad = useRef(true)
 
   useEffect(() => {
-    if (!state?.tradeType) return
+    if (isFirstLoad.current && !!state) {
```

Comment:
I see this is a key change to break the loop.
But what the point of creating `tradeStateFromUrlAtom`? Probably you did it to cache the `tradeStateFromUrl` value, correct?

Reply:
Correct! I thought at first that could be related, since the hook is included in multiple places causing it to be constantly re-evaluated. Decided to add since it was already done 🤷